node/node.go

```diff
@@ -160,6 +160,16 @@ func NewNode(ctx context.Context,
 		return nil, err
 	}
 
+	// The key will be deleted if it existed.
+	// Not checking whether the key is there in case the genesis file was larger than
+	// the max size of a value (in rocksDB for example), which would cause the check
+	// to fail and prevent the node from booting.
+	err = stateDB.Delete(genesisDocKey)
+	if err != nil {
```

Comment:
Just to confirm, `Delete` will not report an error if the key is not present, will it?

Reply:
It seems that for some DBs cometbft-db returns nil if delete returns the `ErrKeyNotFound` error and for some it just forwards the error. Some DBs on the other hand, themsleves do not return  an error. 

I think it is safe to leave this as is here (with the log message), as it will not disrupt the node, but open an issue in the cometbft-db repo to unify this behaviour across all the supported DBs.

Reply:
> to unify this behaviour across all the supported DBs.

... or wait until we chose one and remove the others

Reply:
I think for the scope of this PR it's OK just to leave as a log message if it's not causing any side effects. The scope to adapt or change the DB behaviour is a much bigger scope and agree that we deal with that in the future.